app/hooks/useUpdatePostProvingOpts.ts

```diff
@@ -26,17 +19,13 @@ export default (onFinishHandler: () => void) => {
     setLoading(true);
 
     try {
-      await dispatch(
-        updateProvingOptsAndRestartSmeshing(
-          nonces,
-          threads,
-          postSetupState,
-          isSmeshingStarted
-        )
-      );
+      eventsService.updateProvingOpts({
+        nonces,
+        threads,
+      });
     } catch (error: any) {
       captureReactException(error);
-      // error handles by ErrorBoundary, here we should just finish loading
+      // error handles by ErrorBoundary, we should just finish loading
       setLoading(false);
       return;
     }
```

Comment:
Don't you miss `setLoading(false)` for the good case? 🤔 
I think you can just remove `return` and move `setLoading(false)` to the bottom of the function :-p